org.eclipse.xtext.web.example.statemachine/src/org/eclipse/xtext/web/example/statemachine/formatting2/StatemachineFormatter.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright (c) 2015, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.web.example.statemachine.formatting2;
+
+import org.eclipse.xtext.formatting2.AbstractJavaFormatter;
+import org.eclipse.xtext.formatting2.IFormattableDocument;
+import org.eclipse.xtext.formatting2.regionaccess.ISemanticRegion;
+import org.eclipse.xtext.web.example.statemachine.services.StatemachineGrammarAccess;
+import org.eclipse.xtext.web.example.statemachine.statemachine.Command;
+import org.eclipse.xtext.web.example.statemachine.statemachine.Condition;
+import org.eclipse.xtext.web.example.statemachine.statemachine.Event;
+import org.eclipse.xtext.web.example.statemachine.statemachine.Signal;
+import org.eclipse.xtext.web.example.statemachine.statemachine.State;
+import org.eclipse.xtext.web.example.statemachine.statemachine.Statemachine;
+import org.eclipse.xtext.web.example.statemachine.statemachine.Transition;
+
+import com.google.inject.Inject;
+
+public class StatemachineFormatter extends AbstractJavaFormatter {
+
+	@Inject
+	private StatemachineGrammarAccess ga;
+
+	protected void format(Statemachine statemachine, IFormattableDocument doc) {
+		for (Signal signal : statemachine.getSignals()) {
+			doc.format(signal);
+			doc.append(signal, it -> it.setNewLines(1, 1, 2));
+		}
+		for (State state : statemachine.getStates()) {
+			doc.format(state);
+			doc.append(state, it -> it.setNewLines(1, 1, 2));
+		}
+	}
+
+	protected void format(Signal signal, IFormattableDocument doc) {
+		ISemanticRegion signalKeyword = regionFor(signal).keyword(ga.getSignalAccess().getSignalKeyword_1());
+		doc.prepend(signalKeyword, it -> it.oneSpace());
```

Comment:
Do we want to have delegate methods on the super class for the hidden region formatters?
```doc.prepend(signalKeyword, this::oneSpace);``` ?

Reply:
cc @ArneDeutsch